plugins/guests/redhat/cap/rhn_register.rb

```diff
@@ -0,0 +1,121 @@
+module VagrantPlugins
+  module GuestRedHat
+    module Cap
+      class RhnRegister
+        # Test that the machine is already registered
+        def self.rhn_register_registered?(machine)
+          true if machine.communicate.execute('/usr/sbin/rhn_check', sudo: true)
+        rescue
+          false
+        end
+
+        # Test that we have rhn installed
+        def self.rhn_register(machine)
+          machine.communicate.test('/usr/sbin/rhn_check --version', sudo: true) &&
+            machine.communicate.test('/usr/sbin/rhnreg_ks --version', sudo: true)
+        end
+
+        # Register the machine using 'rhnreg_ks' command, config is (Open)Struct
+        def self.rhn_register_register(machine, ui)
+          rhn_register_upload_certificate(machine, ui) if machine.config.registration.ca_cert
+          rhn_register_server_url(machine, ui) if machine.config.registration.serverurl
+          command = "rhnreg_ks #{configuration_to_options(machine.config.registration)}"
+          machine.communicate.execute("cmd=$(#{command}); if [ \"$?\" != \"0\" ]; then echo $cmd | grep 'This system is already registered' || (echo $cmd 1>&2 && exit 1) ; fi", sudo: true)
+        end
+
+        # Unregister the machine using 'rhn_unregister.py' resource script
+        def self.rhn_register_unregister(machine)
+          tmp = '/tmp/rhn_unregister'
+          username = machine.config.registration.username
+          password = machine.config.registration.password
+
+          machine.communicate.tap do |comm|
+            # No api call can be made without username/password
+            if username && password
+              serverurl = machine.config.registration.serverurl
+              # Generate the api url
+              serverurl = serverurl.sub(/XMLRPC$/, 'rpc/api')
+              comm.sudo("rm -f #{tmp}", error_check: false)
+              comm.upload(resource('rhn_unregister.py'), tmp)
+              comm.sudo("python #{tmp} -u #{username} -p #{password} -s #{serverurl}")
+              comm.sudo("rm -f #{tmp}")
+            end
+            # guest still "thinks" it is a part of RHN network until systemdid file is removed
+            comm.sudo('rm -f /etc/sysconfig/rhn/systemid')
+          end
+        end
+
+        # Return required configuration options for rhn register
+        def self.rhn_register_credentials(_)
```

Comment:
I would prefer you to follow the styling we currently use rather than introducing a new one. Its better to do such changes separately and for the whole project. e.g. Introduce a PR that changes all method definition to use underscore if params are not being used `_`.